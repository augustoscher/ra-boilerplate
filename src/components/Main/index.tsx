import * as S from './styles'

const Main = () => (
  <S.Wrapper>
    <S.Logo
      src="/img/logo.svg"
      alt="Image de um atomo e React Avançado escrito ao lado"
    />
    <S.Title>React Avançado</S.Title>
    <S.Description>
      TypeScript. ReactJS, NextJS e StyledComponents
    </S.Description>
    <S.Ilustration
      src="/img/hero-illustration.svg"
      alt="Desenvolvedor sentado na cadeira de frente para a tela"
    />
  </S.Wrapper>
)

export default Main
