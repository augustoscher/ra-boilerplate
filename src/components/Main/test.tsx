import { render, screen } from '@testing-library/react'

import Main from './index'

describe('<Main />', () => {
  it('match snapshot', () => {
    const { container } = render(<Main />)
    expect(container).toMatchSnapshot()
  })

  it('shoule render the heading', () => {
    render(<Main />)

    expect(
      screen.getByRole('heading', { name: /react avan√ßado/i })
    ).toBeInTheDocument()
  })
})
