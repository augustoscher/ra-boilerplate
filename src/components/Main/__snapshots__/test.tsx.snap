// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> match snapshot 1`] = `
<div>
  <main
    class="styles__Wrapper-sc-18tj86z-0 jZITAN"
  >
    <img
      alt="Image de um atomo e React Avançado escrito ao lado"
      class="styles__Logo-sc-18tj86z-1 cnjhQu"
      src="/img/logo.svg"
    />
    <h1
      class="styles__Title-sc-18tj86z-2 jcjSPU"
    >
      React Avançado
    </h1>
    <h2
      class="styles__Description-sc-18tj86z-3 bmqGsn"
    >
      TypeScript. ReactJS, NextJS e StyledComponents
    </h2>
    <img
      alt="Desenvolvedor sentado na cadeira de frente para a tela"
      class="styles__Ilustration-sc-18tj86z-4 lcSWhC"
      src="/img/hero-illustration.svg"
    />
  </main>
</div>
`
